### ISSUE TITLE
Use closure compiler for build

### ISSUE THREAD

**Issue by osdevisnot** [Polymer/lit-html#717]
It seems current release is weighing in at just over 3k minified & gzipped. Without looking at the outcome, I am tempted to think that using closure compiler for build would help reducing the size even further. If there is an interest, I can probably give it a shot.

**Comment by Lodin** [comment]
Doesn't Closure Compiler [transpile ES6 down to ES5](https://github.com/google/closure-compiler/wiki/ECMAScript6)?

**Comment by osdevisnot** [comment]
Looks like that page is little old. Closure recently added support for ECMASCRIPT_2015 (or es6). Also documented here -- https://github.com/google/closure-compiler/wiki/Flags-and-Options

**Comment by Lodin** [comment]
@osdevisnot Thanks for the link!

Well, regarding usage of the Closure Compiler. AFAIK, `lit-html` is distributed without any minification leaving all processing (transpilation to ES5, minification, polyfilling, etc.) for the end users.

**Comment by osdevisnot** [comment]
@Lodin Precisely, The fact that the distribution is left to end user currently makes it hard to use lit-html directly in browsers. 

Browsers currently do not support bare imports, which means users will need to reply on module bundlers and build system they enforce.

Even if we discount closure compilation in favor of terser, we should atleast have bundled version in distribution IMO.

**Comment by osdevisnot** [comment]
btw, it seems this was discussed before in https://github.com/Polymer/lit-element/issues/140

**Comment by justinfagnani** [comment]
Unless the size is _significantly_ less than something more common in the standard Node world, I don't think it's work the trouble. I'd rather have a realistic number representing what common setups with Rollup and other bundlers would give.
